### ISSUE TITLE
Files stuck in "completing" state

### ISSUE THREAD

**Issue by thany** [meganz/MEGAsync#464]
When uploading hundreds/thousands of files, after a while, several may get stuck in `completing` state. They will never actually complete. Not even when I pause & resume the uploads. Other uploads are happily going, so my internet connection is just fine & dandy.

I would have to completely exit the client, and restart it. Then the `completing` files are gone. But are they actually uploaded? It's hard to figure out since the files in question are disappeared from MEGAsync, and they are among thousands of other files.

So, instead of `completing`, I would like to see what the status *actually* is. Whatever it means, put that there, because `completing` is a very confusing term. It could be "verifying that the file exists on the server" or something.

To recap:

1. Files get stuck in `completing`. They shouldn't get stuck. They should complete.
2. The term `completing` doesn't properly describe what it's actually doing, or trying to do.

Screenshot:

![Capture](https://user-images.githubusercontent.com/152227/86353542-b1458c80-bc67-11ea-8768-ec9696b4d4ab.PNG)

**Comment by jumoog** [comment]
i had the same problem. if you use the sync function change the extension to .jpg. Once its fully uploaded change the extension back. The file rename will be synced as well.

**Comment by jumoog** [comment]
The Installation was broken. Megasync can't create a preview and stucks on `completing`

**Comment by thany** [comment]
It is a new installation for me.

And besides, the installation shouldn't get into a broken state; it should keep working as intended. So even though reinstalling *might* be a workaround, the bug is not gone and still needs fixing.

**Comment by brandyballoon** [comment]
This issue is still present.  I started using MEGA a couple of days ago and several times when right-clicking a folder and selecting "Upload to MEGA" a few of the files are stuck on "completing."  I've left it overnight and they're still there hours later.  Exiting the client and restarting it makes them disappear.  It's very annoying having to take a screenshot of the transfer status and then manually check that each of the affected files did in fact successfully transfer to the server.
![image](https://user-images.githubusercontent.com/110278271/216736806-6ee30f92-d768-45c1-8500-56881d0a60f7.png)

**Comment by thany** [comment]
2.5 years later, and not even a clarification of what `completing` means exactly.

Who can answer this?

**Comment by mattw-mega** [comment]
"Completing" is when the file data has been uploaded to a MEGA file server, and the corresponding Node is being created in your account.  Analogous to making a directory entry in your local filesystem, after actually copying file data.  Requests to the servers to create Nodes are sent in batches along with other types of requests, and have to wait for prior batches to complete first.  So that's one possible reason for delays there.

**Comment by brandyballoon** [comment]
> "Completing" is when the file data has been uploaded to a MEGA file server, and the corresponding Node is being created in your account. Analogous to making a directory entry in your local filesystem, after actually copying file data. Requests to the servers to create Nodes are sent in batches along with other types of requests, and have to wait for prior batches to complete first. So that's one possible reason for delays there.

Thanks for the info.  Once stuck at completing, they stay there.  Hundreds of new transfers will go through in seconds without waiting for the stuck ones to complete first.  A bit more info for you if it helps...  After exiting and restarting the sync client (which makes the stuck ones disappear from the list) I've noticed that the affected files are now duplicated on the server and if I look at the folder in the web portal it warns of the duplicates and I can choose which one to keep.  The options are to rename or keep the newest version.  Both files have the same modification time but selecting keep newest successfully deletes one of them.

**Comment by mattw-mega** [comment]
Hi @brandyballoon , for the case you describe, it sounds like the requests to make those new nodes were sent to the severs, and they did eventually get executed, though you exited MEGAsync in the meantime so it doesn't know about that.   When you restart it, there is a race going on between receiving the updates about those new nodes, and the sync deciding to upload those files again (because it's working on the old snapshot of how your account was - it will be updated but maybe not before it already started a new upload of that same file).   That's how you end up with the duplicates.     These problems are all fixed in our upcoming new 5.0 version which you can try out if you like at #770, or in the meantime please just be patient and let the uploads complete before restarting.    thanks

**Comment by brandyballoon** [comment]
Thanks I might give that a go.

When you say be patient, how many hours are you suggesting?  What if I need to turn my computer off?  They really do just sit there as if nothing at all is happening.  If I upload another folder, all the files might transfer successfully and the original stuck ones from a previous occasion are still there.  One of more files from the new folder may also get stuck on "completing."  It looks to me like some kind of handshake problem where the transfer has actually completed but the client is still waiting for the confirmation that never comes.

**Comment by mattw-mega** [comment]
Hi @brandyballoon , you certainly should not be waiting hours.  A few minutes at most in normal operation.  There can be longer delays if there is some really huge operation going on in the account, such as another client (webclient, maybe) moving a million nodes to the trash.  In that case, the MEGAsync node creations have to wait until that is complete.    But if you are seeing transfers in completing state for hours, then as you suggest it might be that the GUI didn't notice the node completion.  For that sort of thing, you can restart and there shouldn't be any issues as the node and sync databases are likely still updated properly.   You should be fine to restart  so long as the sync isn't still busy uploading other files.   All those issues should be resolved though when we release the changes from #770.  thanks

**Comment by brandyballoon** [comment]
It seems to be happening to about 1 or 2 files out of every 1000.  I had one stuck at "completing" out of a batch of 720 yesterday and leaving it overnight didn't help.  I'm just noticing it a lot because I'm in the process of transferring all my data from onedrive.  In all cases, the affected files either were uploaded properly, or duplicated.  In no cases were the files missing, which is reassuring.  Oh and this is using the right-click -> upload to mega feature, not a synced folder, if that makes any difference.

Thanks again for the response.  I can live with it until the next release as I've finished the bulk transfer now.  I'm reluctant to try 5.0 on my real data at this time because it's alpha.

**Comment by thany** [comment]
> Hi @brandyballoon , you certainly should not be waiting hours. A few minutes at most in normal operation.

Even a few minutes seems excessive. I'm on 1Gb fiberoptic, and if something doesn't work instantly it's the remote being slow, as a general rule.

> There can be longer delays if there is some really huge operation going on in the account, such as another client (webclient, maybe) moving a million nodes to the trash. In that case, the MEGAsync node creations have to wait until that is complete.

Why wait? Also that's kind of an extreme situation. I doubt that's what people do as regularly as they report these "stuck on completing" type of issues.

> But if you are seeing transfers in completing state for hours, then as you suggest it might be that the GUI didn't notice the node completion. For that sort of thing, you can restart and there shouldn't be any issues as the node and sync databases are likely still updated properly.

Why is the GUI not simply re-requesting from the server "yo, is this file completed yet? I'm still waiting for your confirmation"

But the deeper problem lies in the fact that this *can* happen. Why does such an acknowledgement from the server get lost in the first place? A problem like that may be indicative of a deeper problem, which may, at some point, cause a more severe malfunction.

**Comment by 6d657263757279** [comment]
I am having precisely the same issue. Both the mobile app and the desktop app have completely shit the bed. Not only does anything barely even load half the time, but my mobile app is never "caught up" and the desktop app refuses to perform any transfers, getting stuck on the "Completed..." stage and progressing no further. I have tried everything from logging out to completely re-installing. Tried different Wi-Fi connections, etc to no avail.

**Comment by rl-mega** [comment]
Please re-open the ticket if the issue persists with the latest release.
